index.js

```diff
@@ -1,285 +0,0 @@
-    [{ground: rock,x:0,y:1050},{ground: rock,x:50,y:1050},{ground: rock,x:100,y:1050},{ground: rock,x:150,y:1050},{ground: rock,x:200,y:1050}, {ground: rock,x:250,y:1050},  {ground: rock,x:300,y:1050},  {ground: rock,x:350,y:1050},  {ground: rock,x:400,y:1050},  {ground: rock,x:450,y:1050},  {ground: rock,x:500,y:1050},{ground: rock,x:550,y:1050},  {ground: rock,x:600,y:1050}],
-    [{ground: gold,x:0,y:1100},{ground: gold,x:50,y:1100},{ground: gold,x:100,y:1100},{ground: gold,x:150,y:1100},{ground: gold,x:200,y:1100}, {ground: gold,x:250,y:1100},  {ground: gold,x:300,y:1100},  {ground: gold,x:350,y:1100},  {ground: gold,x:400,y:1100},  {ground: gold,x:450,y:1100},  {ground: gold,x:500,y:1100},{ground: gold,x:550,y:1100},  {ground: gold,x:600,y:1100}]
-
-]
-}
-*/
-function makeMap() {
-    // random map, but I want first 3-4 layers to be empty and bottom two to be rock and gold.
-    //to make the first layers empty, change the 0 in the second forloop to be j= 3, then it starts at the third layer
-    const row = 13;
-    const col = 25;
-    mapArr = [];
-  
-    for (let i = 1; i < col; i++) {
-       if(!i==0){
-            mapArr.push([]) 
-        }
-      mapArr.push([]);
-      
-      if (i<col-2){
-      for (let j = 0; j < row; j++) { 
-        let tempObj = {ground: randomMaterial(),x:j*50 ,y:i*50};
-        mapArr[i].push(tempObj); }
-   
-     }else if (i == col-2){
-        for (let g = 0; g < row; g++) { 
-            let tempObj = {ground: rock,x:g*50 ,y:i*50};
-            mapArr[i].push(tempObj);}
-     }else{ 
-        for (let k = 0; k < row; k++) { 
-            let tempObj = {ground: gold,x:k*50 ,y:i*50};
-            mapArr[i].push(tempObj);
-            
-        }
-        }
-    }
-  
-    return mapArr;
-  }
-// makeMap/ mapArr to be changed for  generateMap/ matrix at line 56, , 166, 180, 219,220  
-
- 
-function draw(){ 
-    background(220);
-
-   if (showStart){
-    image(startScreen,0,0,0,1000);
-   }
-     else if(gameRunning && goldCount==0 ){
-    textSize(32);
-    text(`Fuel: ${fuel}`,500,35);
-    if(ironCount<5){
-        text(`Iron Collected: ${ironCount}`,50,35);
-    }else {
-        text(`Ironhack activated!`,50,35);
-    }
-    fill(51);
-    
- for(let i=0;i<mapArr.length;i++){
-        let tempArr = mapArr[i];
-        for (let j=0; j<tempArr.length;j++){
-             image(tempArr[j].ground,tempArr[j].x,tempArr[j].y);
-        }
-    }
-
- image (playerNorm,player.x,player.y);
-
-   if (fuel < 0 && !goldCount>=1){
-        resetButton = false;
-        gameRunning = false;
-    }
-    //out of fuel
-    }else if(showStart==false &&gameRunning== false && goldCount==0){
-    image(gameOverScreen,0,0,0,1000);
-    const button = document.querySelector('#restart');
-     button.style.cssText= `visibility: visible`;
-    }
-    // win screen
-    else if(goldCount>=1){
```

Comment:
this if, else if block etc. could be done with a switch. Also you could have a game state variable that you assign to the different states for the different screens. This way this code becomes more readable as well and you can drop the 
// win screen etc. comments. :)